src/test/java/org/broadinstitute/consent/http/db/DatasetDAOTest.java

```diff
@@ -888,6 +888,25 @@ void testFindAllDatasets() {
     assertTrue(insertedDatasetIds.containsAll(foundDatasetIds));
   }
 
+  @Test
+  void testFindAllDatasetIds() {
+    List<Dataset> datasetList = IntStream.range(1, 5).mapToObj(i -> {
+      Dataset dataset = insertDataset();
+      return dataset;
+    }).toList();
+
+    List<Integer> datasetIds = datasetDAO.findAllDatasetIds();
+    assertFalse(datasetIds.isEmpty());
+    assertEquals(datasetList.size(), datasetIds.size());
+    // Assert that the two lists of ids have all of the same values
+    List<Integer> insertedDatasetIds = datasetList.stream().map(Dataset::getDataSetId).toList();
+    assertTrue(datasetIds.containsAll(insertedDatasetIds));
+    assertTrue(insertedDatasetIds.containsAll(datasetIds));
+    // Assert the list order was preserved in the query
+    assertEquals(datasetList.get(0).getDataSetId(), datasetIds.get(0));
+    assertEquals(datasetList.get(datasetList.size() - 1).getDataSetId(), datasetIds.get(datasetIds.size() - 1));
```

Comment:
I believe all of these assertions could be consolidated thusly:

```suggestion
    List<Integer> insertedDatasetIds = IntStream.range(1, 5).mapToObj(i -> {
      Dataset dataset = insertDataset();
      return dataset.getDataSetId();
    }).toList();
    List<Integer> datasetIds = datasetDAO.findAllDatasetIds();

    assertThat(datasetIds, contains(insertedDatasetIds.toArray()));
```

This matcher checks that the two lists of IDs have exactly the same values in the same order, no more and no less.

Docs: https://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/Matchers.html#contains(E...)

Reply:
Great suggestion, thank you!